### ISSUE TITLE
Gettext.dngettext raise FunctionClauseError in error_helpers.ex with Ecto 2.0 beta

### ISSUE THREAD

**Issue by kenta-polyglot** [phoenixframework/phoenix#1592]
### Environment

* Elixir version (elixir -v): 1.2.3
* Phoenix version (mix deps): 1.1.4
* NodeJS version (node -v): v5.5.0
* NPM version (npm -v): 3.5.3
* Operating system: Mac OS X Yosemite 10.10.5

### Expected behavior

no error.

### Actual behavior

** (exit) an exception was raised:
    ** (FunctionClauseError) no function clause matching in Gettext.dngettext/6
        (gettext) lib/gettext.ex:524: Gettext.dngettext(ForEctoUpgrade.Gettext, "errors", "can't be blank", "can't be blank", nil, %{})
        (my_project) web/views/error_helpers.ex:13: MyProject.ErrorHelpers.error_tag/2

When validation failed, this error always raised after upgrading to newest Ecto 2.0 beta. (source is from Github, not hex) 

This error is influenced by Ecto 2.0's newer commit.
https://github.com/elixir-lang/ecto/commit/77ebf2027c24e667805225f5e38a9667b6ff48b0

5th parameter of Gettext.dngettext must be integer, but nil passed.

I temporarily avoid this error like below:
```elixir
  def translate_error({msg, opts}) do
    Gettext.dngettext(ForEctoUpgrade.Gettext, "errors", msg, msg, opts[:count] || 1, opts)
  end
```

**Comment by whatyouhide** [comment]
This should be solved once #1589 is merged.

**Comment by josevalim** [comment]
Yup, dup of #1589. You can adopt the solution we chose there!
